cdap-examples/ClicksAndViews/src/test/java/co/cask/cdap/examples/clicksandviews/ClicksAndViewsMapReduceTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.examples.clicksandviews;
+
+import co.cask.cdap.api.dataset.lib.PartitionDetail;
+import co.cask.cdap.api.dataset.lib.PartitionKey;
+import co.cask.cdap.api.dataset.lib.PartitionedFileSet;
+import co.cask.cdap.test.ApplicationManager;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.cdap.test.MapReduceManager;
+import co.cask.cdap.test.StreamManager;
+import co.cask.cdap.test.TestBase;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.apache.twill.filesystem.Location;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests that a MapReduce job can process data from two different Streams and perform a reduce-side join across
+ * the data in the two Streams.
+ */
+public class ClicksAndViewsMapReduceTest extends TestBase {
+  private static final Joiner TAB_JOINER = Joiner.on("\t");
+
+  // have views with id [0,5], all for the same adId
+  private static final ImmutableList<String> VIEWS =
+    ImmutableList.of(createView(0, 1461219010, 2157, "http://www.google.com", "lu=fQ9qHjLjFg3qi3bZiuz", "62.128.93.36"),
+                     createView(1, 1461265001, 2157, "http://www.google.co.uk", "lu=8fsdggknea@ASJHlz", "21.612.39.63"),
+                     createView(2, 1461281958, 2157, "http://www.yahoo.com", "name=Mike", "212.193.252.52"),
+                     createView(3, 1461331879, 2157, "http://www.amazon.com", "name=Matt", "1.116.135.146"),
+                     createView(4, 1461348738, 2157, "http://www.t.co", "name=Nicholas; Httponly", "89.141.94.158"),
+                     createView(5, 1461349158, 2157, "http://www.linkedin.com", "lo=Npa0jbIHGloMnx75", "69.75.87.114"));
+
+  private static final ImmutableList<Integer> CLICKS = ImmutableList.of(1, 2, 5);
+
+  private static final ImmutableSet<String> JOINED_VIEWS;
+  static {
```

Comment:
don't need a static block right?  If it cannot be instantiated in place, just use ``@BeforeSubmit``?